.github/workflows/build-runner.yml

```diff
@@ -0,0 +1,64 @@
+---
+"on":
+  pull_request:
+    branches:
+      - "**"
+    paths:
+      - "Dockerfile"
+      - .github/workflows/build-runner.yml
+  push:
+    branches:
+      - master
+    paths:
+      - "Dockerfile"
+      - .github/workflows/build-runner.yml
+
+name: Runner
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build ${{ matrix.name }}
+    strategy:
+      matrix:
+        include:
+          - name: actions-runner
+            dockerfile: Dockerfile
+    env:
+      RUNNER_VERSION: 2.274.1
+      DOCKER_VERSION: 19.03.13
+      DOCKERHUB_USERNAME: ${{ github.repository_owner }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+
+      - name: Build Container Image
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          docker buildx build \
+            --platform linux/amd64,linux/arm64 \
+            --tag ${DOCKERHUB_USERNAME}/${{ matrix.name }}:v${RUNNER_VERSION} \
+            --tag ${DOCKERHUB_USERNAME}/${{ matrix.name }}:latest \
+            -f ${{ matrix.dockerfile }} .
+      - name: Login to GitHub Docker Registry
+        run: echo "${DOCKERHUB_PASSWORD}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+        if: ${{ github.event_name == 'push' }}
+        env:
+          DOCKERHUB_USERNAME: ${{ github.repository_owner }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+
+      - name: Build and Push Container Image
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          docker buildx build \
+            --build-arg RUNNER_VERSION=${RUNNER_VERSION} \
+            --build-arg DOCKER_VERSION=${DOCKER_VERSION} \
+            --platform linux/amd64,linux/arm64 \
+            --tag ${DOCKERHUB_USERNAME}/${{ matrix.name }}:v${RUNNER_VERSION} \
+            --tag ${DOCKERHUB_USERNAME}/${{ matrix.name }}:latest \
+            -f ${{ matrix.dockerfile }} . --push
```

Comment:
`--push` nice 👍  - I didn't know you could `docker build` and at the same time push ☺️

Reply:
You have to specify either --load or --push with buildx otherwise it doesn't do anything 👍